File: packages/algo-py/src/algo_00.py
def most_frequent(arr):
    ret = None
    counter = {}
    max_count = -1
    for n in arr:
        counter.setdefault(n, 0)
        counter[n] += 1
        if counter[n] > max_count:
            max_count = counter[n]
            ret = n
    return ret

File: packages/algo-ts/src/algo_00.ts
export function mostFrequent(arr: number[]): number {
  let ret = -1;
  let maxCount = -1;
  const counter = new Map<number, number>();
  for (const n of arr) {
    let count = 1;
    if (!counter.has(n)) {
      counter.set(n, count);
    } else {
      count = counter.get(n)!;
      count++;
    }
    if (count > maxCount) {
      maxCount = count;
      ret = n;
    }
  }
  return ret;
}
